Lib/test/test_uuid.py

```diff
@@ -313,63 +315,62 @@ def test_getnode(self):
         node2 = self.uuid.getnode()
         self.assertEqual(node1, node2, '%012x != %012x' % (node1, node2))
 
-    def _setup_for_pickle(self):
-        orig_uuid = sys.modules.get('uuid')
-        sys.modules['uuid'] = self.uuid
-
-        def restore_uuid_module():
-            if orig_uuid is not None:
-                sys.modules['uuid'] = orig_uuid
-            else:
-                del sys.modules['uuid']
-        self.addCleanup(restore_uuid_module)
-
     def test_pickle_roundtrip(self):
-        self._setup_for_pickle()
+        def pickle_roundtrip(obj, protocol):
+            return pickle.loads(pickle.dumps(obj, protocol=protocol))
+        copiers = [
+            ('copy.copy', copy.copy),
+            ('copy.deepcopy', copy.deepcopy),
+        ] + [
+            (f'pickle round-trip with protocol={protocol}',
+             partial(pickle_roundtrip, protocol=protocol))
+            for protocol in range(pickle.HIGHEST_PROTOCOL + 1)
+        ]
 
-        u = self.uuid.UUID('12345678123456781234567812345678')
-        self.assertEqual(u, pickle.loads(pickle.dumps(u)))
+        with support.swap_item(sys.modules, 'uuid', self.uuid):
+            for copy_func_name, copy_func in copiers:
+                with self.subTest(copy_func_name=copy_func_name):
+                    for is_safe in self.uuid.SafeUUID:
+                        u = self.uuid.UUID('d82579ce6642a0de7ddf490a7aec7aa5',
+                                           is_safe=is_safe)
+                        self.assertEqual(u, copy_func(u))
 
     def test_unpickle_previous_python_versions(self):
-        self._setup_for_pickle()
-
-        u = self.uuid.UUID('12345678123456781234567812345678')
+        u = self.uuid.UUID('d82579ce6642a0de7ddf490a7aec7aa5')
 
         # Python 2.7 protocol 0-2 pickles of u
         py27_pickles = [
-            b'ccopy_reg\n_reconstructor\np0\n(cuuid\nUUID\np1\nc__builtin__\nob'
-            b'ject\np2\nNtp3\nRp4\n(dp5\nS\'int\'\np6\nL24197857161011715162171'
-            b'839636988778104L\nsb.',
-            b'ccopy_reg\n_reconstructor\nq\x00(cuuid\nUUID\nq\x01c__builtin__\n'
-            b'object\nq\x02Ntq\x03Rq\x04}q\x05U\x03intq\x06L2419785716101171516'
-            b'2171839636988778104L\nsb.',
-            b'\x80\x02cuuid\nUUID\nq\x00)\x81q\x01}q\x02U\x03intq\x03\x8a\x10xV'
-            b'4\x12xV4\x12xV4\x12xV4\x12sb.',
+            b'ccopy_reg\n_reconstructor\n(cuuid\nUUID\nc__builtin__\nobject\nNt'
+            b'R(dS\'int\'\nL287307832597519156748809049798316161701L\nsb.',
+            b'ccopy_reg\n_reconstructor\n(cuuid\nUUID\nc__builtin__\nobject\nNt'
```

Comment:
It would look clearer if add comments between items. E.g.:
```python
            ...
            # Python 2.7, protocol 2
            b'\x80\x02cuuid\nUUID\n)\x81}U\x03int\x8a\x11\xa5z\xecz\nI\xdf}\xde'
            b'\xa0Bf\xcey%\xd8\x00sb.',
            # Python 3.6, protocol 0
            ...
```